lib/Refactoring/AddExplicitCodableImplementation.cpp

```diff
@@ -12,183 +12,258 @@
+  }
 
-  AddCodableContext(ExtensionDecl *Decl)
-      : DC(Decl), StartLoc(Decl->getBraces().Start),
-        Protocols(getAllProtocols(Decl->getExtendedNominal())),
-        Range(Decl->getMembers()){};
+  AccessLevel getContainerAccess() const {
+    if (auto *nominal = dyn_cast<NominalTypeDecl>(IDC->getDecl())) {
+      return nominal->getFormalAccess();
+    }
+    if (auto *extension = dyn_cast<ExtensionDecl>(IDC->getDecl())) {
+      return extension->getDefaultAccessLevel();
+    }
 
-  AddCodableContext() : DC(nullptr), Protocols(), Range(nullptr, nullptr){};
+    llvm_unreachable("unknown IterableDeclContext");
+  }
 
-  static AddCodableContext
-  getDeclarationContextFromInfo(ResolvedCursorInfoPtr Info);
+  /// Get the left brace location of the type-or-extension decl.
+  SourceLoc getLeftBraceLoc() const {
+    if (auto *nominal = dyn_cast<NominalTypeDecl>(IDC->getDecl())) {
+      return nominal->getBraces().Start;
+    }
+    if (auto *extension = dyn_cast<ExtensionDecl>(IDC->getDecl())) {
+      return extension->getBraces().Start;
+    }
 
-  void printInsertionText(ResolvedCursorInfoPtr CursorInfo, SourceManager &SM,
-                          llvm::raw_ostream &OS);
+    llvm_unreachable("unknown IterableDeclContext");
+  }
 
-  bool isValid() { return StartLoc.isValid() && conformsToCodableProtocol(); }
+  /// Get the token location where the text should be inserted after.
+  SourceLoc getInsertStartLoc() const {
+    // Prefer the end of elements.
+    for (auto *member : llvm::reverse(IDC->getParsedMembers())) {
+      if (isa<AccessorDecl>(member) || isa<VarDecl>(member))
+        continue;
+      return member->getEndLoc();
+    }
 
-  SourceLoc getInsertStartLoc();
-};
+    // After the starting brace if empty.
+    return getLeftBraceLoc();
+  }
 
-SourceLoc AddCodableContext::getInsertStartLoc() {
-  SourceLoc MaxLoc = StartLoc;
-  for (auto Mem : Range) {
-    if (Mem->getEndLoc().getOpaquePointerValue() >
-        MaxLoc.getOpaquePointerValue()) {
-      MaxLoc = Mem->getEndLoc();
+  std::string getBaseIndent() const {
+    SourceManager &SM = IDC->getDecl()->getASTContext().SourceMgr;
+    SourceLoc startLoc = getInsertStartLoc();
+    StringRef extraIndent;
+    StringRef currentIndent =
+        Lexer::getIndentationForLine(SM, startLoc, &extraIndent);
+    if (startLoc == getLeftBraceLoc()) {
+      return (currentIndent + extraIndent).str();
+    } else {
+      return currentIndent.str();
     }
   }
-  return MaxLoc;
-}
 
-/// Walks an AST and prints the synthesized Codable implementation.
-class SynthesizedCodablePrinter : public ASTWalker {
-private:
-  ASTPrinter &Printer;
-
-public:
-  SynthesizedCodablePrinter(ASTPrinter &Printer) : Printer(Printer) {}
-
-  MacroWalking getMacroWalkingBehavior() const override {
-    return MacroWalking::Arguments;
-  }
+  /// Returns true if the member of IDC is a part of the synthesized Codable.
+  bool shouldPrintMember(ValueDecl *VD) const {
```

Comment:
Am I missing something or is this never called?

Reply:
Thanks for spotting! This was a leftover from earlier development.